src/v3/test/integration/byol.test.tsx

```diff
@@ -0,0 +1,125 @@
+    jest.spyOn(global, 'localStorage', 'get').mockReturnValue({
+      length: 0,
+      clear: () => jest.fn(),
+      getItem: () => null,
+      setItem: () => jest.fn(),
+      key: () => null,
+      removeItem: () => jest.fn(),
+    });
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('loading custom language', () => {
+    const server: SetupServerApi = setupServer();
+    beforeAll(() => {
+      server.listen();
+    });
+
+    beforeEach(() => {
+      server.use(
+        rest.get('http://localhost:1234/mocks/labels/json/login_foo.json', async (_, res, ctx) => res(
+          ctx.status(200),
+          ctx.json(loginFooBundle),
+        )),
+        rest.get('http://localhost:1234/mocks/labels/json/country_foo.json', async (_, res, ctx) => res(
+          ctx.status(200),
+          ctx.json(countryFooBundle),
+        )),
+      );
+    });
+
+    afterEach(() => {
+      server.resetHandlers();
+    });
+
+    afterAll(() => {
+      server.close();
+    });
+
+    it('should load custom language with assets.baseUrl and assets.languages set and with navigator.languages containing language', async () => {
+      const navigatorLanguagesSpy = jest.spyOn(global, 'navigator', 'get');
+      navigatorLanguagesSpy.mockReturnValue(
+        { languages: ['foo'] } as unknown as Navigator,
+      );
+      const { container, findByText } = await setup({
+        mockResponse,
+        widgetOptions: {
+          assets: {
+            baseUrl: 'http://localhost:1234/mocks',
+            languages: ['foo'],
+          },
+        },
+      });
+      await waitFor(async () => {
+        await findByText(/Set up foo authentication/);
+        await findByText(/Foonited States/);
+      });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should load custom language with "language" option defined, and assets.baseUrl is set to a path containing the language assets', async () => {
+      const { container, findByText } = await setup({
+        mockResponse,
+        widgetOptions: {
+          language: 'foo',
+          assets: { baseUrl: 'http://localhost:1234/mocks' },
+        },
+      });
+      await waitFor(async () => {
+        await findByText(/Set up foo authentication/);
+        await findByText(/Foonited States/);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('loading default language', () => {
+    it('should not load custom language with "language" option set', async () => {
+      const { container, findByText } = await setup({
+        mockResponse,
+        widgetOptions: { language: 'foo' },
+      });
+      await waitFor(async () => { await findByText(/Set up phone authentication/); });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should not load by default, with assets.baseUrl set to a path containing the language assets with navigator.languages containing language', async () => {
```

Comment:
same here about clarifying the test description. it looks like here the reason is that there is not a `language/languages` set in widgetOptions